pkg/metrics/metric/metric.go

```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package metric
+
+import (
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// WithMetadata is the interface implemented by any metric defined in this package. These typically embed existing
+// prometheus metric types and add additional metadata to the definitions which can be used to generate proper metric
```

Comment:
Capitalize product names unless they're specifically lower-cased; for example, Prometheus vs gRPC.
```suggestion
// Prometheus metric types and add additional metadata to the definitions which can be used to generate proper metric
```